src/includes/class-alma-wc-payment-gateway.php

```diff
@@ -783,19 +782,32 @@ private function render_payments_timeline( $eligibility ) {
 	}
 
 	/**
-	 * Validate Fields.
+	 * Validate payment fields.
 	 *
 	 * @return bool
 	 */
 	public function validate_fields() {
+		$error_msg = __( 'There was an error processing your payment.<br>Please try again or contact us if the problem persists.', 'alma-gateway-for-woocommerce' );
+
 		$alma_fee_plan = $this->checkout_helper->get_chosen_alma_fee_plan();
+
 		if ( ! $alma_fee_plan ) {
+			wc_add_notice( $error_msg, 'error' );
+
 			return false;
 		}
+
+		$is_alma_payment = $this->checkout_helper->is_alma_payment_method();
+
+		if ( ! $is_alma_payment ) {
```

Comment:
```suggestion
		if ( ! $this->checkout_helper->is_alma_payment_method() ) {
```
unnecessary variable, you could inline it